Fix button rendering bug in directory entry admin listing

chigby opened the pull request:
This fixes a bug on the directory entry admin listings where if the latest scan result for a directory entry in the listing does not exist, then the button creation hook would raise an exception and the admin page would not render.  I've fixed it by only adding that button if the latest result actually is present.

I was running into this while doing some testing locally, but I don't see why it wouldn't happen in production eventually, either.

chigby committed:
Only create latest scan link if latest scan exists

This fixes a bug where if the latest result did not exist, then the
button creation hook would raise an exception and the admin page
would not render.  I've fixed it by only adding that button if the
latest result actually is present.

SaptakS commented on directory/wagtail_hooks.py:
```diff
@@ -76,19 +76,23 @@ def add_bundle_stats_button(page, page_perms, is_parent=False):
         )
     ]
+    latest = page.get_live_result()
+    if latest and latest.pk:
```
Makes sense. I remember I added a similar check for a new directory page where latest wouldn't be there. Might have missed this one at that time.

chigby committed:
Ignore safety check for cryptography version <3.3

We can't upgrade because pshtt depends on an old version of sslyze
which does not support the latest cryptography.

This is the information from safety:

39252: cryptography <3.3 resolved (3.2.1 installed)!
Cryptography 3.3 no longer allows loading of finite field Diffie-Hellman parameters of less than 512 bits in length. This change is to conform with an upcoming OpenSSL release that no longer supports smaller sizes. These keys were already wildly insecure and should not have been used in any application outside of testing.

chigby committed:
Ensure filter test does not fail based on randomness

This change excludes all languages and countries from the
`other_instance` (the one designed not to show up in the filters under
test).  Without setting languages and countries to 0, they will be
assigned two random ones on creation, which will sometimes coincide
with the countries we're testing in the filter and cause the test to
fail.
